client/components/site-verticals-suggestion-search/index.jsx

```diff
@@ -157,72 +151,52 @@ export class SiteVerticalsSuggestionSearch extends Component {
 	};
 
 	render() {
-		const { translate, placeholder, autoFocus, shouldShowPopularTopics } = this.props;
-		const showPopularTopics = shouldShowPopularTopics( this.state.searchValue );
+		const {
+			translate,
+			placeholder,
+			autoFocus,
+			shouldShowPopularTopics,
+			isVerticalSearchPending,
+		} = this.props;
+		const showPopularTopics = shouldShowPopularTopics( this.props.searchValue );
+
 		return (
 			<>
+				<QueryVerticals searchTerm={ this.props.searchValue.trim() } />
+				<QueryVerticals searchTerm={ DEFAULT_VERTICAL_KEY } limit={ 1 } />
```

Comment:
What would be the best way to offer a `debounce` here, baked into the `<QueryVertical />` component itself?

Just looking at the HTTP requests we're firing off, and we could probably skip over a few for the fast typers out there.

Reply:
Right. If we want to add a debouncing behavior, I'd say baking into the query component itself would be ideal. I'd also make it configurable with a prop, e.g. `debounceThreshold` or `debounceTime`, so that it can still do live request if needed, e.g. in https://github.com/Automattic/wp-calypso/pull/31747 .